core-android/src/main/java/com/uber/sdk/android/core/auth/SsoDeeplink.java

```diff
@@ -131,24 +132,31 @@ private Uri createSsoUri() {
      */
     @Override
     public boolean isSupported() {
+        Pair<SupportedAppType, PackageInfo> installedUberAppPackage = appProtocol.getInstalledUberApp(activity);
+        if (installedUberAppPackage == null) {
+            return false;
+        }
 
-        PackageInfo packageInfo = null;
-        for (String installedPackage : AppProtocol.UBER_PACKAGE_NAMES) {
-            if (PackageManagers.isPackageAvailable(activity, installedPackage)) {
-                packageInfo = PackageManagers.getPackageInfo(activity, installedPackage);
-                break;
-            }
+        SupportedAppType supportedApp = installedUberAppPackage.first;
+        int minimumVersion;
+        if (UBER == supportedApp) {
+            minimumVersion = MIN_UBER_RIDES_VERSION_SUPPORTED;
+        } else if (UBER_EATS == supportedApp) {
+            minimumVersion = MIN_UBER_EATS_VERSION_SUPPORTED;
+        } else {
+            return false;
         }
 
-        return (packageInfo != null)
-                && appProtocol.validateMinimumVersion(activity, packageInfo, MIN_VERSION_SUPPORTED)
+        PackageInfo packageInfo = installedUberAppPackage.second;
+        return packageInfo != null
+                && appProtocol.validateMinimumVersion(activity, packageInfo, minimumVersion)
                 && appProtocol.validateSignature(activity, packageInfo.packageName);
```

Comment:
These checks should probably get moved up one level or we should return all installed apps (In a list) and check them both here. ie, what if rider is < the version but eats is not? What if rider is a malicious build but eats is valid?

Reply:
That's a great point.  I went back and forth on moving more or less of the min version logic up to AppProtocol.  Will refactor this.

Reply:
@tyvsmith 
I've updated the `AppProtocol#isInstalled` and `SsoDeeplink` to support multiple apps and their minimum versions.  I've ensured that all checks include signature validation.